examples/ios/objc/Puzzle/Views/RLMStartView.m

```diff
@@ -0,0 +1,335 @@
+{
+    if (self.superview) {
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillShow:) name:UIKeyboardWillShowNotification object:nil];
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillHide:) name:UIKeyboardWillHideNotification object:nil];
+    }
+    else {
+        [[NSNotificationCenter defaultCenter] removeObserver:self name:UIKeyboardWillShowNotification object:nil];
+        [[NSNotificationCenter defaultCenter] removeObserver:self name:UIKeyboardWillHideNotification object:nil];
+    }
+
+    if (self.logoView == nil) {
+        self.logoView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"RealmLogo"]];
+        [self addSubview:self.logoView];
+    }
+
+    if (self.tableView == nil) {
+        self.tableView = [[UITableView alloc] initWithFrame:(CGRect){CGPointZero, (CGSize){520.0f, 54.0f*3.0f}} style:UITableViewStylePlain];
+        self.tableView.backgroundColor = [UIColor clearColor];
+        self.tableView.autoresizingMask = UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleLeftMargin
+                                            | UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleBottomMargin;
+        self.tableView.dataSource = self;
+        self.tableView.delegate = self;
+        self.tableView.scrollEnabled = NO;
+        self.tableView.rowHeight = 54.0f;
+        self.tableView.layer.shadowRadius = 20.0f;
+        self.tableView.layer.shadowOpacity = 0.1f;
+        self.tableView.layer.shadowPath = [UIBezierPath bezierPathWithRoundedRect:self.tableView.frame cornerRadius:20.0f].CGPath;
+        self.tableView.clipsToBounds = NO;
+        [self.contentView addSubview:self.tableView];
+        
+        //Suppress the separator line at the bottom
+        UIView *fillerView = [[UIView alloc] initWithFrame:CGRectMake(0,0,1,1)];
+        self.tableView.tableFooterView = fillerView;
+    }
+    
+    if (self.topCellBackgroundView == nil) {
+        self.topCellBackgroundView = [[UIImageView alloc] initWithImage:[RLMStartView cellBackgroundImageBottom:NO]];
+        self.topCellBackgroundView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    }
+    
+    if (self.bottomCellBackgroundView == nil) {
+        self.bottomCellBackgroundView = [[UIImageView alloc] initWithImage:[RLMStartView cellBackgroundImageBottom:YES]];
+        self.bottomCellBackgroundView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    }
+    
+    if (self.hostNameField == nil) {
+        self.hostNameField = [self newTextField];
+        self.hostNameField.placeholder = @"localhost";
+        
+        NSString *hostName = [[NSUserDefaults standardUserDefaults] stringForKey:kRLMPuzzleHostNameKey];
+        if (hostName.length > 0) {
+            self.hostNameField.text = hostName;
+        }
+        else if (kLocalIPAddress.length > 0) {
```

Comment:
The constant generated by the Xcode build phase is actually named `kIPAddress`...